DatadogFlags/Sources/Feature/ExposureLogger.swift

```diff
@@ -0,0 +1,75 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+internal protocol ExposureLogging {
+    func logExposure(
+        at date: Date,
+        for flagKey: String,
+        assignment: FlagAssignment,
+        context: FlagsEvaluationContext
+    )
+}
+
+internal final class ExposureLogger: ExposureLogging {
+    private struct Exposure: Hashable {
+        let targetingKey: String
+        let flagKey: String
+        let allocationKey: String
+        let variationKey: String
+    }
+
+    private let featureScope: any FeatureScope
+    private var loggedExposures: Set<Exposure> = []
```

Comment:
Should we be concerned that the cache can grow without bound?

For comparison, `AssignmentCache` in [Eppo's iOS SDK](https://github.com/Eppo-exp/eppo-ios-sdk/blob/efdd1b19f66878d82a9558bc19883924800503ef/Sources/eppo/AssignmentCache.swift) didn't have any mechanisms for controlling the cache size either and it hasn't caused any issues.